phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4896,6 +4954,20 @@ else if (metaProperties.getImmutableStorageSchemeProp() != table.getImmutableSto
                 metaProperties.setNonTxToTx(true);
             }
         }
+
+        if (metaProperties.getViewTTL() != null) {
+            if (table.getType() == PTableType.INDEX) {
```

Comment:
Shouldn't the condition be `if(table.getType() != PTableType.VIEW)`?

Reply:
evaluateStmtProperties() is typically called by alterIndex and addColumn, but no harm in making it more general.